### PR TITLE
Fix RigidbodyWing for quaternion

### PR DESCRIPTION
**Pull request by xuhao1**
I try to fix the problem that when rotation is quaternion we can't use rigidbodywing...
And in my simulation this piece of code would run.
As for I am not familiar with Drake Source Code(and it lack document for me to understand...Everything is guess from source). Plz make sure the mathematics on line 541 and line 542 works.

### CODE REVIEW

**Comment by tkoolen** [comment]
I think you're missing a qd on line 444.

File: drake/systems/plants/@RigidBodyWing/RigidBodyWing.m [hunk 1]
```diff
@@ -534,7 +535,14 @@ function revertEnvVars()
         dforce = sparse(numel(force),2*nq)*q(1); % q(1) is for taylorvar
         dforce((frame.body_ind-1)*6+1:frame.body_ind*6,:) = [dbody_forcedq,dbody_forcedqd];
         dforce = reshape(dforce,6,[]);
```
Comment (tkoolen) [inline drake/systems/plants/@RigidBodyWing/RigidBodyWing.m]: so this is being completely overwritten on the next line?

File: drake/systems/plants/@RigidBodyWing/RigidBodyWing.m [hunk 2]
```diff
@@ -534,7 +535,14 @@ function revertEnvVars()
...
+        dforce = sparse(6*getNumBodies(manip),getNumStates(manip));
+        % As for tmp has non zero value on only row 1 2 3
+        % I give it value to dforce only 1-3
+        tmp = ([dbody_forcedq,dbody_forcedqd]);
+        dforce((frame.body_ind-1)*6+1:frame.body_ind*6,1:3) = tmp(:,1:3);
```
Comment (tkoolen) [inline drake/systems/plants/@RigidBodyWing/RigidBodyWing.m]: What is your reasoning behind this?

File: drake/systems/plants/@RigidBodyWing/RigidBodyWing.m [hunk 3]
```diff
@@ -658,14 +666,17 @@ function revertEnvVars()
...
       %     velocity of the wing in world coordinates
       %   </pre>
-      
+      options.in_terms_of_qdot = false;
       if (nargout>1)
         
-        [~, J, dJ] = forwardKin(manip,kinsol,kinframe,zeros(3,1));
-        wingvel_world_xyz = J*qd; % assume still air. Air flow over the wing
+        %[~, J, dJ] = forwardKin(manip,kinsol,kinframe,zeros(3,1));
+        [x,J_geometric,dJ_geometric_dq] = forwardKin(manip,kinsol,kinframe,zeros(3,1),options);
```
Comment (tkoolen) [inline drake/systems/plants/@RigidBodyWing/RigidBodyWing.m]: What we call a geometric Jacobian (and also in the literature) is a matrix that maps from v to twist, so I think choosing a different name would be better.

File: drake/systems/plants/@RigidBodyWing/RigidBodyWing.m [hunk 4]
```diff
@@ -658,14 +666,17 @@ function revertEnvVars()
...
+        [~,dx_dq] = forwardKin(manip,kinsol,kinframe,zeros(3,1),struct('in_terms_of_qdot',true));
+        
+        wingvel_world_xyz = J_geometric*qd; % assume still air. Air flow over the wing
```
Comment (tkoolen) [inline drake/systems/plants/@RigidBodyWing/RigidBodyWing.m]: J_geometric is in terms of v, but now you're postmultiplying qd.

**Comment by tkoolen** [comment]
I think RigidBodyWing.computeWingVelocity is the first thing that needs to change; it should truly accept v instead of still doing stuff in terms of qd, as is currently the case. It should also compute gradients w.r.t. v instead of qd.

File: drake/systems/plants/@RigidBodyWing/RigidBodyWing.m (hunk 3 above)
Comment (hongkai-dai) [inline drake/systems/plants/@RigidBodyWing/RigidBodyWing.m]: I think I should be blamed for this one. I called it geometric Jacobian in RigidBodyPropellor. m

**Comment by xuhao1** [comment]
Aha... I am a little busy these days, so do your guys has soluction on this?

**Pull request closed without merging**